src/components/Api.js

```diff
@@ -0,0 +1,127 @@
+export class Api {
+  constructor({ baseUrl, headers, contentType }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+    this._contentType = contentType;
+  }
+
+  async getInitialCards() {
+    try {
+      const res = await fetch(`${this._baseUrl}/cards`, {
+        method: "GET",
+        headers: {
+          authorization: this._headers,
+        },
+      });
+
+      if (!res.ok) {
+        throw new Error(`HTTP error! Status: ${res.status}`);
+      }
+
+      return await res.json(); // si pasa el error, devuelve la respuesta como JASON (objeto)
+    } catch (err) {
```

Comment:
Needs correction: The errors should be handled outside of the Api class in index.js

Reply:
hello kseniya57! thank you for take your time and do my corrections, could I ask for a suggestion on how to resolve this correction?

Reply:
Hello! You should not handle the errors in Api.js, you should handle them in index.js:

```js
// Api.js
 async getInitialCards() {
      const res = await fetch(`${this._baseUrl}/cards`, {
        method: "GET",
        headers: {
          authorization: this._headers,
        },
      });

      return this._processResponse(res)
}

// index.js

// way 1
api.getInitialCards(..).then(..).catch(..)

// way 2
function getInitialCards() {
  try {
    const cards = await api.getInitialCards(..);
    ..
  } catch (error) {
    ..
  }
}
```

Reply:
I erased the error catch on line 22 and I handle the errors in the Index.js